pkg/provider/aws/provider.go

```diff
@@ -146,9 +162,9 @@ func newClient(ctx context.Context, store esv1beta1.GenericStore, kube client.Cl
 
 	switch prov.Service {
 	case esv1beta1.AWSServiceSecretsManager:
-		return secretsmanager.New(sess, cfg)
+		return secretsmanager.New(sess, cfg, util.IsReferentSpec(prov.Auth))
```

Comment:
```suggestion
		return secretsmanager.New(sess, cfg, false)
```
Otherwise, the SecretStore `Validate()` are handled as `Unknown`, when they should be verified for validity.

I didn't see any case where this could really be set to true... but maybe I'm wrong?